website/docs/tutorials/08-cairo-1-support/02-project-initialization.md

```diff
@@ -14,100 +17,150 @@ protostar init-cairo1 my-project
+and `lib.cairo` files defined.
 
-1. Create a `mod1` directory in the `src` crate
-2. Create `mod1.cairo` alongside this directory. 
-3. Create your source file inside of `mod1` (i.e. `functions.cairo` or any suitable name). Define your code here.
-4. Declare the source file/files in `mod1.cairo`. The file contents should look like this (assuming you have `functions.cairo` from the previous step):
 ```
-mod functions;
+my_project/
+├── package1/
+│   ├── src/
+│   │   ├── contracts/
+│   │   │   └── hello_starknet.cairo
+│   │  ...
+│   │   ├── contracts.cairo
+│   │   └── lib.cairo
+│   └── cairo_project.toml
+├── package2/
+│   ├── src/
+│   │   ├── contracts/
+│   │   │   └── other_contract.cairo
+│   │  ...
+│   │   ├── contracts.cairo
+│   │   └── lib.cairo
+│   └── cairo_project.toml
+...
+└── protostar.toml
 ```
-5. Declare the module in the root of the module tree - `lib.cairo`. After adding, the file contents should look like this:
+
+Make sure `[crate_roots]` are correctly defined.
+
+```toml title="package1/cairo_project.toml"
+[crate_roots]
+package1 = "src"
 ```
-mod main;
-mod mod1;
+
+```toml title="package2/cairo_project.toml"
+[crate_roots]
+package2 = "src"
 ```
-6. You can import the symbols from `functions.cairo` in tests. For example, in `test_main.cairo`:
+
+Define each contract in the `[contracts]` section of the protostar.toml and each package
+in the `linked-librares`
+
+```toml title="protostar.toml"
+# ...
+linked-libraries = ["package1", "package2"]
+
+[contracts]
+hello_starknet = ["package1"]
+other_contract = ["package2"]
 ```
-use src::mod1::functions::three;
 
+### Testing multi-contract projects
+
+For example, to test function `returns_two` defined in the `package1/business_logic/utils.cairo` write
+
+```cairo title="Example test"
 #[test]
-fn test_numbers() {
-    assert(3 == three(), 'three() == 3');
+fn test_returns_two() {
+    assert(package1::business_logic::utils::returns_two() == 2, 'Should return 2');
 }
 ```
 
-## The protostar.toml
-Apart from the usual things you can find in `protostar.toml`, there is a `linked-libarires` entry which is used to find cairo 1 modules in tests and building.
-This makes it possible for you to include other modules from your dependencies if they are correct cairo 1 modules (with their own module definition and `cairo_project.toml`).
+Or using the `use path:to::mod` syntax
 
-```
-[project]
-protostar-version = "0.9.2"
-lib-path = "lib"
-linked-libraries = ["src"]
+```cairo title="Example test
+use package1::business_logic::utils::returns_two;
 
-[contracts]
+#[test]
+fn test_returns_two() {
+    assert(returns_two() == 2, 'Should return 2');
+}
 ```
 
-:::warning
-`[contracts]` section is not usable right now, since protostar can't build cairo 1 contracts yet 
-:::
\ No newline at end of file
+Make sure that the path::to:the::module is correct for your package structure.
```

Comment:
Can we add info how to do integration tests as well?

Reply:
What do you mean exactly? Perhaps let's just link to the testing page of the docs here so we don't duplicate knowledge.

Reply:
That's fine